plugins/win-wasapi/win-wasapi.cpp

```diff
@@ -1138,17 +1138,42 @@ bool WASAPISource::ProcessCaptureData()
 			return false;
 		}
 
+		// ToDo figure out if we need to reset capture here
+		if (flags & AUDCLNT_BUFFERFLAGS_TIMESTAMP_ERROR) {
+			blog(LOG_ERROR, "[WASAPISource::ProcessCaptureData]"
+					" Timestamp error!");
+			capture->ReleaseBuffer(frames);
+			return false;
+		}
+
+		if (flags & AUDCLNT_BUFFERFLAGS_DATA_DISCONTINUITY) {
+			/* libobs should handle discontinuities fine. */
+			blog(LOG_DEBUG, "[WASAPISource::ProcessCaptureData]"
+					" Discontinuity flag is set.");
+		}
+
+		if (flags & AUDCLNT_BUFFERFLAGS_SILENT) {
+			blog(LOG_DEBUG, "[WASAPISource::ProcessCaptureData]"
+					" Silent flag is set.");
+
+			/* buffer size = frame size * number of frames
+			 * frame size = channels * sample size
+			 * sample size = 4 bytes (always float per InitFormat) */
+			uint32_t requiredBufSize = 4 * speakers * frames;
+			if (silence.size() < requiredBufSize)
+				silence.resize(requiredBufSize);
+
+			buffer = silence.data();
+		}
+
 		obs_source_audio data = {};
 		data.data[0] = (const uint8_t *)buffer;
```

Comment:
Is this cast necessary? Maybe you can remove it while you're here.

Reply:
Both this and the case for `frames` do seem redundant, yeah.